packages/plugin-data/src/api/data/tree/exportApi.ts

```diff
@@ -0,0 +1,607 @@
+                    const id: string = record[key];
+                    const relatedObject = this.referenceRegistry[relTo];
+                    if (relatedObject) {
+                      const ref: string = relatedObject[id];
+                      // If ref is not found, then leave intact because we may not have processed
+                      // this parent fully. We'll go back through the sObject tree
+                      // later and replace the id with a reference.
+                      return ref ? `@${ref}` : id;
+                    } else {
+                      // again, this will just be the id for now and replaced with a ref later.
+                      return id;
+                    }
+                  } else {
+                    // TODO: what to do if ref not found?
+                    const recordId: string = record['Id'];
+                    this.logger.error(`Reference ${relTo} not found for ${key}.  Skipping record ${recordId}.`);
+                  }
+                }
+              } else {
+                // not a relationship field, simple key/value insertion
+                return record[key];
+              }
+            }
+
+            return null;
+          })
+          .then((processedAttribute) => {
+            if (processedAttribute !== null) {
+              treeRecord[key] = processedAttribute;
+            }
+
+            return Promise.resolve(null);
+          })
+          .catch((e) => {
+            throw e;
+          });
+      }
+
+      return Promise.resolve(null);
+    });
+  }
+
+  // Get sObject description and cache for given object type
+  private async loadMetadata(objectName: string): Promise<any> {
+    if (!describe[objectName]) {
+      const sObject = this.org.getConnection().sobject(objectName);
+      describe[objectName] = await sObject.describe();
+    }
+
+    return describe[objectName];
+  }
+
+  private isQueryResult(metadata: any, fieldName: string): any {
+    return metadata.childRelationships.some((cr: any) => cr.relationshipName === fieldName);
+  }
+
+  private isSpecificTypeWithMetadata(metadata: any, fieldName: string, fieldType: string): boolean {
+    for (let i = 0, fld; i < metadata.fields.length; i++) {
+      fld = metadata.fields[i];
+      if (fld.name.toLowerCase() === fieldName.toLowerCase()) {
+        if (fld.type.toLowerCase() === fieldType.toLowerCase()) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  private getRelationshipFieldName(metadata: any, parentName: string): string {
+    let result = '';
+    metadata.fields.some((field: any) => {
+      if (field.type === 'reference') {
+        for (const refTo of field.referenceTo) {
+          if (refTo === parentName) {
+            result = field.name;
+            return true;
+          }
+        }
+      }
+
+      return false;
+    });
+
+    if (!result) {
+      const metadataName: string = metadata.name;
```

Comment:
absolute nit: but why have a variable for this? just print `${metadata.name}`

Reply:
because of linting.  There is a rule about a string template using a non-string.  The type of metadata.name is any, so I can either do `${metadata.name as string}` or the code above (or better typing overall, but that's another effort).